assets/src/hooks/useDetour.ts

```diff
@@ -126,7 +129,11 @@ export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
 
   return {
     /** The current state of the detour machine */
-    state,
+    state:
+      snapshot.matches({ "Detour Drawing": "Editing" }) &&
+      !snapshot.matches({ "Detour Drawing": "Share Detour" })
```

Comment:
Nit: Is it possible for `snapshot.matches({ "Detour Drawing": "Share Detour" })` to be true if `snapshot.matches({ "Detour Drawing": "Editing" })` is true?

If not, then we don't need the `!snapshot.matches({ "Detour Drawing": "Share Detour" })` part of the condition.

Reply:
It "MUST NOT" be true, in the words of [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) 😅 
When I was writing this I thought I came up with a pretty good way to do this but upon reflection and seeing it again, it's a bit silly.